command.go

```diff
@@ -850,42 +852,64 @@ func (c *Command) HasSubCommands() bool {
 	return len(c.commands) > 0
 }
 
+// Determine if a command is available as a non-help command (this includes all
+// non deprecated/hidden commands)
+func (c *Command) IsAvailableCommand() bool {
+
+	// a command is 'available' if it is runnable and is not deprecated/hidden
+	return c.Runnable() && (len(c.Deprecated) == 0) && !c.Hidden
```

Comment:
unnecessary `()` around all of your `len() !=` tests through here. Otherwise squash and I'll merge.